warehouse/accounts/services.py

```diff
@@ -75,6 +78,19 @@ def check_password(self, userid, password):
 
         return True
 
+    def create_user(self, user):
+        if self.find_userid(user.username):
+            raise UserAlreadyExists(
+                "User with username {0} already exists!".format(user.username)
+            )
+        self.db.add(user)
+        self.db.commit()
+
+    def update_user(self, user):
```

Comment:
Similarly to above, I think this would be better off if it accepted a userid and a list of things to change, like:

```python
def update_user(self, userid, **changes):
    user = self.get_user(userid)
    for attr, value in changes.items():
        setattr(user, attr, value)
    return user  # Maybe? Not sure if we should return the user here or just let it alone
```